File: backend/src/main/java/org/kmurygin/healthycarbs/mealplan/dto/IngredientDTO.java
package org.kmurygin.healthycarbs.mealplan.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.kmurygin.healthycarbs.mealplan.IngredientCategory;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class IngredientDTO {
    private Long id;
    private String name;
    private String unit;
    private Double caloriesPerUnit;
    private Double carbsPerUnit;
    private Double proteinPerUnit;
    private Double fatPerUnit;
    private IngredientCategory category;
}

File: frontend/src/app/core/models/dto/ingredient.dto.ts
import type {IngredientCategory} from "../enum/ingredient-category.enum";
import type {AllergenDto} from "@core/models/dto/allergen.dto";
import type {UserDto} from "@core/models/dto/user.dto";

export interface IngredientDto {
  readonly id: number;
  readonly name: string;
  readonly unit: string;
  readonly caloriesPerUnit: number;
  readonly carbsPerUnit: number;
  readonly proteinPerUnit: number;
  readonly fatPerUnit: number;
  readonly category: IngredientCategory;
  readonly allergens: readonly AllergenDto[];
  readonly author: UserDto | null;
}
